src/components/AgreementSections/FormInfo.js

```diff
@@ -48,14 +58,39 @@ export default class FormInfo extends React.Component {
       newState.isPerpetualValues = [{ value: null, label: '' }, ...data.isPerpetualValues];
     }
 
+    if (data.reasonForClosureValues.length + 1 !== state.reasonForClosureValues.length) {
+      newState.reasonForClosureValues = [{ value: null, label: '' }, ...data.reasonForClosureValues];
+    }
+
+    // Check if status is closed and, if so, change state to allow reasonForClosure field
+    if (values.agreementStatus === statuses.CLOSED) {
+      newState.isClosed = true;
+    } else {
+      newState.isClosed = false;
+    }
+
     if (Object.keys(newState).length) return newState;
 
     return null;
   }
 
-  render() {
-    const { agreementStatusValues, isPerpetualValues, renewalPriorityValues } = this.state;
+  setWarnings = ({ values }) => {
+    let warning;
+
+    if (values.reasonForClosure && values.agreementStatus !== statuses.CLOSED) {
+      warning = (
+        <div data-test-warn-clear-reason-for-closure>
+          <FormattedMessage id="ui-agreements.warn.clearReasonForClosure" />
+        </div>
+      );
+    } else {
+      warning = undefined;
```

Comment:
This `else` case isn't necessary. If you `let foo;`, then `foo` is `undefined` from the start.